subsys/net/ip/dhcpv4.c

```diff
@@ -1069,11 +1072,40 @@ static enum net_verdict net_dhcpv4_input(struct net_conn *conn,
 	return NET_DROP;
 }
 
+static void iface_handler(struct net_mgmt_event_callback *cb,
+			  u32_t mgmt_event, struct net_if *iface)
+{
+	if (iface->config.dhcpv4.state == NET_DHCPV4_INIT) {
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_IF_DOWN) {
+		NET_DBG("Interface %p going down", iface);
+
+		iface->config.dhcpv4.attempts = 0;
+		iface->config.dhcpv4.state = NET_DHCPV4_RENEWING;
+		NET_DBG("enter state=%s",
+			net_dhcpv4_state_name(iface->config.dhcpv4.state));
+	} else if (mgmt_event == NET_EVENT_IF_UP) {
+		NET_DBG("Interface %p coming up", iface);
+
+		if (iface->config.dhcpv4.state == NET_DHCPV4_RENEWING) {
```

Comment:
(following my comment about k_delayed_work_submitg below) this should check for INIT as well then.